hypershift-operator/controllers/nodepool/nodepool_controller_test.go

```diff
@@ -2622,6 +2622,309 @@ func TestTaintsToJSON(t *testing.T) {
 	}
 }
 
+func TestSetPerformanceProfileStatus(t *testing.T) {
+	controlPlaneNamespace := "clusters-hostedcluster01"
+	userClustersNamespace := "clusters"
+	nodePoolName := "hostedcluster01"
+
+	testCases := []struct {
+		name                         string
+		PerformanceProfileStatusCM   *corev1.ConfigMap
+		expectedConditions           map[string]hyperv1.NodePoolCondition
+		hasPerformanceProfileApplied bool
+	}{
+
+		{
+			name:                         "No Performance profile applied",
+			PerformanceProfileStatusCM:   &corev1.ConfigMap{},
+			expectedConditions:           map[string]hyperv1.NodePoolCondition{},
+			hasPerformanceProfileApplied: false,
+		},
+
+		{
+			name: "Performance profile is available",
+			PerformanceProfileStatusCM: &corev1.ConfigMap{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ConfigMap",
+					APIVersion: "v1",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "perfprofile-" + nodePoolName + "-status",
+					Namespace: controlPlaneNamespace,
+					Labels: map[string]string{
+						"hypershift.openshift.io/nto-generated-performance-profile-status": "true",
+						"hypershift.openshift.io/nodePool":                                 nodePoolName,
+						"hypershift.openshift.io/performanceProfileName":                   nodePoolName,
+					},
+					Annotations: map[string]string{
+						"hypershift.openshift.io/nodePool": nodePoolName,
+					},
+				},
+				Data: map[string]string{
+					"status": `conditions:
+- type: Available
+  status: "True"
+  reason: ""
+  message: ""
+  lastheartbeattime: "2024-04-18T06:55:45Z"
+  lasttransitiontime: "2024-04-18T06:55:45Z"
+- type: Upgradeable
+  status: "True"
+  reason: ""
+  message: ""
+  lastheartbeattime: "2024-04-18T06:55:45Z"
+  lasttransitiontime: "2024-04-18T06:55:45Z"
+- type: Progressing
+  status: "False"
+  reason: ""
+  message: ""
+  lastheartbeattime: "2024-04-18T06:55:45Z"
+  lasttransitiontime: "2024-04-18T06:55:45Z"
+- type: Degraded
+  status: "False"
+  reason: ""
+  message: ""
+  lastheartbeattime: "2024-04-18T06:55:45Z"
+  lasttransitiontime: "2024-04-18T06:55:45Z"
+tuned: openshift-cluster-node-tuning-operator/openshift-node-performance-performance
+runtimeclass: performance-performance`,
```

Comment:
In order to make the tests more readables, maybe you can create some helper functions to fill the tests data fields, it will be more readable.